optas/builder.py

```diff
@@ -58,49 +60,52 @@ def __init__(self, T, robots=[], tasks=[], optimize_time=False, derivs_align=Fal
 
         # Ensure T is sufficiently large
         if not derivs_align:
-
             # Get max time deriv
             all_time_derivs = []
             for m in self._models:
                 all_time_derivs += m.time_derivs
             max_time_deriv = max(all_time_derivs)
 
             # Check T is large enough
-            Tmin = max_time_deriv+1
+            Tmin = max_time_deriv + 1
             assert T >= Tmin, f"{T=} is too low, it should be at least {Tmin}"
 
         model_names = [m.get_name() for m in self._models]
         is_unique_names = len(model_names) == len(set(model_names))
         assert is_unique_names, "each model should have a unique name"
 
-        # Setup decision variables
+        # Setup containers for decision variables, parameters, cost terms, ineq/eq constraints
         self._decision_variables = SXContainer()
-        for model in self._models:
-            for d in model.time_derivs:
-                n = model.state_name(d)
-                if model.T is None:
-                    t = T-d if not derivs_align else T
-                else:
-                    t = model.T
-                self.add_decision_variables(n, model.dim, t)
-
-        if optimize_time:
-            self.add_decision_variables('dt', T-1)
-
-        # Setup containers for parameters, cost terms, ineq/eq constraints
         self._parameters = SXContainer()
         self._cost_terms = SXContainer()
         self._lin_eq_constraints = SXContainer()
         self._lin_ineq_constraints = SXContainer()
         self._ineq_constraints = SXContainer()
         self._eq_constraints = SXContainer()
 
+        # Setup decision variables and parameters
+        for model in self._models:
+            for d in model.time_derivs:
+                n_s = model.state_name(d)
+                if model.T is None:
+                    t = T - d if not derivs_align else T
+                else:
+                    t = model.T
```

Comment:
This should be modified (this relates to a comment I made in the `_solve` method (see the one with **NOTE**). Please change to 
```python
for model in self._models:
    if model.T is None:
        model.T = T
    for d in model.time_derivs:
        n_s = model.state_name(d)
        t = model.T - d if not derivs_align else T
```

Reply:
This is actually not needed now (I realized in the above that the dimensions for the zeros array are readily available without making this change). However, I think it would be a good idea to make this change anyway. Makes the code a little cleaner. Also, model.T shouldn't be left as `None`.

Reply:
Done.